includes/Services/SiteGenService.php

```diff
@@ -253,6 +254,260 @@ public static function generate_child_theme( $data ) {
 
 	}
 
+	/**
+	 * Gets the preview homepages
+	 *
+	 * @return array
+	 */
+
+	public static function generate_homepages($site_description, $content_style, $target_audience, $regenerate = false) {
+        // Fetch homepages using the SiteGen::get_home_pages method
+        $home_pages = SiteGen::get_home_pages(
+            $site_description,
+            $content_style,
+            $target_audience,
+            $regenerate
+        );
+
+        // Process the homepages
+        $processed_home_pages = self::process_homepages_response($home_pages);
+
+        // Update the option with processed homepages
+        update_option('nfd-sitegen-homepages', $processed_home_pages);
+
+        return $processed_home_pages;
+    }
+
+	/**
+	 * Updates the list of favourited slugs
+	 *
+	 * @return array
+	 */
+	public static function toggle_favorite_homepage($slug) {
+        $favorites = get_option('nfd-sitegen-favorites', []);
+        $homepages = get_option('nfd-sitegen-homepages', []);
+
+        if (in_array($slug, $favorites)) {
+            $favorites = array_diff($favorites, [$slug]);
+        } else {
+            $favorites[] = $slug;
+        }
+
+        update_option('nfd-sitegen-favorites', $favorites);
+
+        foreach ($homepages as &$homepage) {
+            if ($homepage['slug'] === $slug) {
+                $homepage['isFavourited'] = in_array($slug, $favorites);
+                break;
+            }
+        }
+
+        update_option('nfd-sitegen-homepages', $homepages);
+
+        return ['message' => 'Favorite status updated', 'favorites' => $favorites];
+    }
+
+	/**
+	 * Regenerate previews for favourited homepages
+	 *
+	 * @return array
+	 */
+	public static function handle_favorite_regeneration($regenerateSlug, $regenerateColorPalattes) {
+        $existing_homepages = get_option('nfd-sitegen-homepages', []);
+        $favorite_regenerate_homepage = array_filter($existing_homepages, function ($homepage) use ($regenerateSlug) {
+            return $homepage['slug'] === $regenerateSlug;
+        });
+
+        if (!empty($favorite_regenerate_homepage)) {
+            $processed_homepage = self::process_favorited_regenerate($favorite_regenerate_homepage, $regenerateColorPalattes);
+            $existing_homepages[] = $processed_homepage[0];
```

Comment:
Maybe return the inside array itself and add to `$existing_homepages[]` in one go?